sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Primitives/ServiceBusMessageActions.cs

```diff
@@ -245,7 +245,8 @@ protected ServiceBusMessageActions()
         {
             if (_receiver is ServiceBusSessionReceiver || _sessionEventArgs != null)
             {
-                throw new InvalidOperationException(Resources.CannotLockMessageOnSessionEntity);
+                throw new InvalidOperationException("Messages cannot be locked when working with session-enabled entities. Locks are handled at the session level." +
+                                                    "In order to manually renew the session lock, bind to the 'ServiceBusSessionMessageActions' type and call 'RenewSessionLockAsync'.");
```

Comment:
Should we consider `nameof` for the method and type names?

Reply:
Hmm, I think we are safe here as these are public types.